### PR TITLE
Add label option to hero and section intro components

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Part of https://github.com/github/primer/issues/2504 and https://github.com/github/primer/issues/2498

## List of notable changes:

- **added** new label options to both hero and section intro components
- **updated** documentation examples **but not the anatomy screenshots**
- **updated** all tests

## What should reviewers focus on?

- Verify correctness of visual appearance
- See if anything is missing

## Steps to test:

🔗  Storybook for SectionIntro
🔗  Documentation for SectionIntro
🔗  Storybook for Hero
🔗  Documentation for Hero

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

Hero

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2023-08-01 at 11 31 10

 </td>
<td valign="top">

Screenshot 2023-08-01 at 11 30 50

</td>
</tr>
</table>

Section Intro

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

Screenshot 2023-08-01 at 11 32 25

 </td>
<td valign="top">

Screenshot 2023-08-01 at 11 32 13

</td>
</tr>
</table>

Test coverage

Screenshot 2023-08-01 at 11 22 23

### CODE REVIEW

File: packages/react/src/Hero/Hero.tsx [hunk 1]
```diff
@@ -72,6 +73,18 @@ function HeroDescription({size = '400', weight, children}: PropsWithChildren<Her
   )
 }
 
+type HeroLabelProps = LabelProps & BaseProps<HTMLDivElement>
+
+function HeroLabel({children, ...rest}: PropsWithChildren<HeroLabelProps>) {
+  return (
+    <div>
```
Comment (danielguillan) [inline packages/react/src/Hero/Hero.tsx:80]: Do we need this wrapping `div`?

File: packages/react/src/SectionIntro/SectionIntro.tsx [hunk 2]
```diff
@@ -75,6 +76,18 @@ const _Description = forwardRef(
   },
 )
 
+type SectionIntroLabelProps = LabelProps & BaseProps<HTMLDivElement>
+
+function _Label({children, ...rest}: PropsWithChildren<SectionIntroLabelProps>) {
+  return (
+    <div>
```
Comment (danielguillan) [inline packages/react/src/SectionIntro/SectionIntro.tsx:83]: Same as above, do we need this wrapping `div`?

**Review by danielguillan — APPROVED** [review]
Looks good! I only left a small, non-blocking question.

File: packages/react/src/Hero/Hero.tsx (hunk 1 above)
Comment (rezrah) [reply packages/react/src/Hero/Hero.tsx:80]: Not necessary, no. I put it there mostly for semantics as adjacent elements are all block level. I can remove it 👍

**Pull request merged**